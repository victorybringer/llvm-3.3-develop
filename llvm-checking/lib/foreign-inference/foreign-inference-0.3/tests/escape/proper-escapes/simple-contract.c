
typedef void (*funcType)(int *);

extern int g1;
extern int g2;

void libfunc1(int *p)
{
  g1 = *p;
}

void libfunc2(int *p)
{
  g2 = *p;
}

struct S
{
  int * i;
  funcType f;
};


void init1(struct S *s)
{
  s->i = 0;
  s->f = libfunc1;
}

void init2(struct S *s)
{
  s->i = 0;
  s->f = libfunc2;
}

void testF(int * p, int* junk, struct S *globalS)
{
  libfunc1(junk);
  libfunc2(junk);
  init1(globalS);
  init2(globalS);
  globalS->f(p);
}
