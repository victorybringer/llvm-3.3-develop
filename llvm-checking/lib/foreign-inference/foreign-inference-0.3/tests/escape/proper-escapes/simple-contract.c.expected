fromList [ ("@testF", fromList [(BrokenContractEscape, "%p")]) ]
